### PR TITLE
[WIP] feature/ATC-219

### PR DESCRIPTION
**Pull request by n8rzz**
completes #219 

adds sockets for server and client and begins laying groundwork for multiplayer

@erikquinn opening this now so you can have a peak at what I'm doing with this.  Feel free to make notes, ask questions, and provide feedback whenever you feel the need.

### CODE REVIEW

**Review by erikquinn — COMMENTED** [review]
Some failing tests:

image

Also it doesn't seem to load at the [https://atc-dev-pr-221.herokuapp.com/](https://atc-dev-pr-221.herokuapp.com/) link.

File: tools/tasks/scriptTasks.js [hunk 1]
```diff
@@ -30,6 +32,14 @@ module.exports = function(gulp, config) {
         .pipe(gulp.dest(OPTIONS.DIR.BUILD_SCRIPTS));
```
Comment (erikquinn) [inline tools/tasks/scriptTasks.js]: @n8rzz Should this be `OPTIONS.DIR.BUILD_SCRIPTS_CLIENT`? When I amend it as such, the game at least loads.

Comment (n8rzz) [reply tools/tasks/scriptTasks.js]: you are correct, this (and the failing tests) are updated in #227 .

I'll pull that in for fun, but this PR will be closed soon cause its just an experiment. 😉

**Comment by n8rzz** [comment]
won't merge, closing

**Pull request closed without merging**